neuraxle/base.py

```diff
@@ -667,6 +669,7 @@ def set_train(self, is_train: bool = True):
             :func:`BaseStep.set_train`
         """
         self.is_train = is_train
+        self.apply(method_name='set_train', is_train=is_train, children_only=True)
```

Comment:
SRP seems broken here. The BaseStep should probably not be aware of the existence of children arguments in its own method. I don't know another solution here but there might be something better. 

> "The interface-segregation principle (ISP) states that no client should be forced to depend on methods it does not use."
- https://hackernoon.com/interface-segregation-principle-bdf3f94f1d11

Here it's not a method but an argument. In other comments I've put in the code below, methods seems to be a problem really.